src/app/backend/client/manager.go

```diff
@@ -97,6 +100,38 @@ func (self *clientManager) Config(req *restful.Request) (*rest.Config, error) {
 	return cfg, nil
 }
 
+// ClientCmdConfig creates ClientCmd Config based on authentication information extracted from request.
+// Currently request header is only checked for existence of 'Authentication: BearerToken'
+func (self *clientManager) ClientCmdConfig(req *restful.Request) (clientcmd.ClientConfig, error) {
```

Comment:
If we look at simplified code then cyclomatic complexity is lower because there is no custom logic involved. Result structure is created by calling other methods. Looking at code complexity it would get worse because indeed we are creating additional structure that normally is not needed, however this will still be O(1) as is only involves reassignment of structures. The advantage here is that we can keep authorization related code strictly in one method and if we'll want to support other authorization options then we have to change only 1 method not 2.